pkg/controllermanager/controller/seed/seed_reconcile.go

```diff
@@ -88,7 +88,7 @@ func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		},
 	}
 
-	if _, err := controllerutil.CreateOrUpdate(ctx, gardenClient.Client(), namespace, func() error {
+	if _, err := controllerutil.CreateOrUpdate(ctx, gardenClient.DirectClient(), namespace, func() error {
```

Comment:
Shouldn't it be

```suggestion
	if _, err := controllerutil.CreateOrUpdate(ctx, gardenClient.APIReader(), namespace, func() error {
```

given that `DirectClient()` is already deprecated?

Reply:
I do not get it. `CreateOrUpdate` needs a client supporting `Update` request, but `APIReader()` supports only `Get` and `List`.

Reply:
Looking into https://github.com/gardener/gardener/pull/3641 I think we will have to re-implement the `controllerutil.CreateOrUpdate` to make it work with reader and client. As this is out of the scope of this PR, I removed this commit.